src/fsharp/tast.fs

```diff
@@ -2543,7 +2535,9 @@ and
 and ValData = Val
 and [<NoEquality; NoComparison; StructuredFormatDisplay("{DebugText}")>]
     Val = 
-    { 
+    { /// Mutable for global state
+      mutable val_global_state: CompilerGlobalState option
```

Comment:
This is a problem - it is adding a word to the critical Val data structure, which we have worked very hard to remove every word from.

I will search around for what makes this neccessary